src/commands/login.js

```diff
@@ -46,7 +47,19 @@ const login = async (context, firstTime = true) => {
   const username = await utils.getInput(QUESTION_USERNAME);
   const password = await utils.getInput(QUESTION_PASSWORD, { secret: true });
 
-  const deployKey = (await utils.createCredentials(username, password)).key;
+  let goodResponse;
+  try {
+    goodResponse = await utils.createCredentials(username, password);
+  } catch ({ errText, json: { errors } }) {
+    if (errors[0].startsWith('missing totp_code')) {
+      const code = await utils.getInput(QUESTION_TOTP);
+      goodResponse = await utils.createCredentials(username, password, code);
+    } else {
+      throw new Error(errText);
+    }
+  }
+
+  const { key: deployKey } = goodResponse;
```

Comment:
Maybe this is simpler?

```suggestion
  const deployKey = goodResponse.key;
```